### PR TITLE
RemoveStartEndNoise speeded up

### PR DESCRIPTION
**Pull request by ivandrofly**
Test => http://pastebin.com/LPfeRYei
Speed improved +10 seconds
If see that this method isn't been used for now...

### CODE REVIEW

File: src/Forms/RemoveTextFromHearImpaired.cs [hunk 1]
```diff
@@ -76,31 +76,21 @@ private void FixLargeFonts()
...
-                s = s.Substring(0, s.Length - 4);
-            if (s.EndsWith("</I>") && s.Length > 4)
-                s = s.Substring(0, s.Length - 4);
-            if (s.EndsWith("</U>") && s.Length > 4)
+            // <i> <I> | <b> <B> | <u> <U>
+            if (text.Length > 3 && text[0] == '<' && text[2] == '>' &&
+                ((text[1] & i) == text[1] || (text[1] & b) == text[1] || (text[1] & u) == text[1]))
+            {
+                s = s.Substring(0, 3);
+            }
```
Comment (xylographe) [inline src/Forms/RemoveTextFromHearImpaired.cs]:
```
if (s.Length > 3 && s[0] == '<' && s[2] == '>' && "ibuIBU".Contains(s[1])
{
    s = s.Substring(3)
}
```
Perhaps?

But it won't handle `text = "<i><b>Lorem ipsum dolor sit amet</b></i>"`

File: src/Forms/RemoveTextFromHearImpaired.cs [hunk 2]
```diff
@@ -76,31 +76,21 @@ private void FixLargeFonts()
...
+            // </i> </I> | </b> </B> | </u> </U>
+            int idx = (s.Length - 4);
+            if(s.Length > 4 && s[idx + 2] == '/' && s[idx] == '<' &&
+                 ((text[idx + 3] & i) == text[idx] || (text[idx + 3] & b) == text[idx] || (text[idx + 3] & u) == text[idx]))
                 s = s.Substring(0, s.Length - 4);
```
Comment (xylographe) [inline src/Forms/RemoveTextFromHearImpaired.cs]:
```
if (idx > 0 && s[idx] == '<' && s[idx+1] == '/' && s[idx+3] == '>' && "ibuIBU".Contains(s[idx+2])
{
    s = s.Remove(idx)
}
```

**Comment by ivandrofly** [comment]
@xylographe as you can see I'm trying to avoid using string which will decrease speed

**Comment by ivandrofly** [comment]
@xylographe http://pastebin.com/LPfeRYei which your  tweaks and report the results you got

**Comment by xylographe** [comment]
Sure, optimization is good. But, the functionality of the code should not be altered.

**Pull request closed without merging**